src/theory/quantifiers/sygus/cegis_core_connective.cpp

```diff
@@ -108,7 +110,7 @@ bool CegisCoreConnective::processInitialize(Node conj,
   std::vector<Node> echildren;
   echildren.push_back(d_candidate);
   echildren.insert(echildren.end(), d_vars.begin(), d_vars.end());
-  d_eterm = NodeManager::currentNM()->mkNode(Kind::DT_SYGUS_EVAL, echildren);
+  d_eterm = nodeManager()->mkNode(Kind::DT_SYGUS_EVAL, echildren);
```

Comment:
This class has a node manager as a member, right? All its member functions should use `d_nm` directly.
```suggestion
  d_eterm = d_nm->mkNode(Kind::DT_SYGUS_EVAL, echildren);
```

Reply:
No, the `CegisCoreConnective` class does not have a node manager as a direct member (`d_nm`). Instead, the class inherits from `EnvObj` via `Cegis` and `SygusModule`. The `EnvObj` class includes an associated environment, `d_env`, as a member. Therefore, you can either call `nodeManager()` directly or access it indirectly via `d_env.getNodeManager()`. However, the latter approach might be less desirable.